flexmeasures/data/models/generic_assets.py

```diff
@@ -578,16 +605,24 @@ def create_generic_asset(generic_asset_type: str, **kwargs) -> GenericAsset:
     ).one_or_none()
     if generic_asset_type is None:
         raise ValueError(f"Cannot find GenericAssetType {asset_type_name} in database.")
-    new_generic_asset = GenericAsset(
+
+    new_generic_asset = get_or_create_generic_asset(
         name=kwargs["name"],
         generic_asset_type_id=generic_asset_type.id,
         attributes=kwargs["attributes"] if "attributes" in kwargs else {},
     )
+
+    # GenericAsset(
+    #     name=kwargs["name"],
+    #     generic_asset_type_id=generic_asset_type.id,
+    #     attributes=kwargs["attributes"] if "attributes" in kwargs else {},
+    # )
```

Comment:
In which commit did these changes happen? I can't seem to find argumentation, while I feel it reflects a fundamental change to what the function `create_generic_asset` is doing. Was this an accidental commit, or was it part of solving some problem?

Reply:
It happened https://github.com/FlexMeasures/flexmeasures/pull/770/commits/48fd692ea798a58b8d245a68be6e2bb41fd5d073, but the argumentation is missing indeed, sorry. 

I had to introduce these changes as to create another sensor for the production prices, required for the test `test_battery_solver_day_3`. The class `Market` was creating a generic asset twice (one of `epex_da` and another for `epex_da_production`) and these changes check if an asset has already been created and creates it if it's not found.

Reply:
Probably this can be reverted then once the Market class is removed (hopefully that PR can be merged tomorrow).